bin/propolis-server/src/lib/initializer.rs

```diff
@@ -1212,6 +1234,10 @@ impl MachineInitializer<'_> {
                 MachineInitError::FwcfgInsertFailed("bootorder", e)
             })?;
         }
+        let e820_entry = self.generate_e820()?;
+        fwcfg
+            .insert_named("etc/e820", e820_entry)
+            .map_err(|e| MachineInitError::FwcfgInsertFailed("e820", e))?;
```

Comment:
I don't really see a reason to make this optional, at least. Regardless of offering up an e820 map the guest OS will use UEFI service and get the map that OVMF builds of the system.